src/attributes.jl

```diff
@@ -397,13 +405,64 @@ function submit(model::ModelLike, sub::AbstractSubmittable, args...)
     end
 end
 
+## Submittables
+
 """
-    SettingSingleVariableFunctionNotAllowed()
+    LazyConstraint(callback_data)
 
-Error type that should be thrown when the user [`set`](@ref) the
-[`ConstraintFunction`](@ref) of a [`SingleVariable`](@ref) constraint.
+Lazy constraint `func`-in-`set` submitted as `func, set`. The optimal
+solution returned by [`VariablePrimal`](@ref) will satisfy all lazy
+constraints that have been submitted.
+
+This can be submitted only from the [`LazyConstraint`](@ref). The
+field `callback_data` is a solver-specific callback type that is passed as the
+argument to the feasible solution callback.
 """
-struct SettingSingleVariableFunctionNotAllowed <: Exception end
+struct LazyConstraint{CBDT} <: AbstractSubmittable
+    callback_data::CBDT
+end
+
+"""
+    HeuristicSolution(callback_data)
+
+Heuristically obtained feasible solution. The solution is submitted as
+`variables, values` where `values[i]` gives the value of `variables[i]`,
+similarly to [`set`](@ref).
+
+This can be submitted only from the [`HeuristicCallback`](@ref). The
+field `callback_data` is a solver-specific callback type that is passed as the
+argument to the heuristic callback.
+
+Some solvers require a complete solution, others only partial solutions. It's up
+to you to provide the appropriate one. If in doubt, give a complete solution.
+
+Note that the solver may silently reject the provided solution.
```

Comment:
Gurobi at least provides a mechanism for finding out if the solution was accepted: https://www.gurobi.com/documentation/8.1/refman/c_grbcbsolution.html (see the `objP` argument). This is *very* useful for unit testing, so we may want to think of a way to expose it. Maybe the `HeuristicSolution` can hold a `Ref{SomeEnumType}` where `SomeEnumType` tells you if the solution was accepted, rejected, or unknown?

Reply:
We could also say that it returns the enum. We never said that `submit` always returns nothing, it could something that depends on the `Submittable`